### DISCUSSION TITLE
Document how to generate SEA that does not log experimental warning

### DISCUSSION THREAD

**Discussion by cspotcode** [nodejs/single-executable#60] [Q&A]
How do we generate an SEA that does not log this warning?

`ExperimentalWarning: Single executable application is an experimental feature and might change at any time`

The executable I generate will not be changing at any time if it's version-locked, and users of said application likely won't see any breaking experimental changes.

**Comment by RaisinTen** [comment]
To be able to do this, the [`--no-warnings`](https://nodejs.org/api/cli.html#--no-warnings) CLI option needs to be prebaked into the single-executable. That isn't possible right now. It would be possible once we have a [layout for co-existing SEA resources](https://github.com/nodejs/single-executable/discussions/58), which is being implemented in https://github.com/nodejs/node/pull/47125.

**Reply by cspotcode** [reply]
Ok thanks for the context.  Though I don't actually want to suppress *all* warnings, only the one that is intrinsic to the executable: that it is an SEA.

For example, if I wrap up a cosmiconfig-style CLI tool into a single executable, and someone does `tool --config foo.config.cjs`, and that config file uses some experimental features, I still want those warnings.

**Reply by RaisinTen** [reply]
Fair enough. I think once that PR lands, it would be possible to add an option to the layout for not displaying that warning.

**Reply by RaisinTen** [reply]
I have sent a PR to implement this - https://github.com/nodejs/node/pull/47588
